test/libsolidity/util/ContractABIUtils.cpp

```diff
@@ -245,6 +250,23 @@ bool ContractABIUtils::appendTypesFromName(
 			_dynamicTypes.push_back(ABIType{ABIType::String, ABIType::AlignLeft});
 		}
 	}
+	else if (isFixedPointType(type))
+	{
+		vector<string> fixedTypeParameter;
+		solAssert(boost::starts_with(type, "fixed"), "");
```

Comment:
Instead of all these assertions and length-based checks, you can also replicate the regex here and use match groups to retrieve the numbers.

Another way is to use `TokenTraits::fromIdentifierOrKeyword`

Reply:
Acutally the current implementation is missing `ufixed`

Reply:
Oh yes! And `fixed` is currently not supporting negative values.